### PR TITLE
Use device tree to generate CONFIG_UART_CONSOLE_ON_DEV_NAME defines

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: scripts/extract_dts_includes.py [hunk 1]
```diff
@@ -404,6 +404,20 @@ def extract_single(node_address, yaml, prop, key, prefix, defs, def_label):
...
+  prop = node['props'][key]
+
+  k = convert_string_to_label(key).upper()
+  prop_def[label] = "\"" + prop + "\""
+
+  if node_address in defs:
+    defs[node_address].update(prop_def)
+  else:
+    defs[node_address] = prop_def
+
```
Comment (agross-linaro) [inline scripts/extract_dts_includes.py]: I wonder if the extract_property shouldnt be more generic and use the type to decide what to do about the value.  As it is right now, the extract_single is being called within the extract_property.  If extract single handled the stringification, we'd be ok to just use that.  The only other issue to deal with is the label.

**Review by agross-oss — COMMENTED** [review]
All of it looks fine except for my questions concerning the addition of the extract string property function.

File: scripts/extract_dts_includes.py (hunk 1 above)
Comment (galak) [inline scripts/extract_dts_includes.py]: hmm, need to look at that.  What exactly is extract_property trying to do.  The stringification seems to happen ok in extract_single, but what's going on with the label from the yaml I don't quite understand.

Comment (agross-linaro) [inline scripts/extract_dts_includes.py]: it basically steers the handling of the property.  The question is whether or not we should really be keying off the yaml to figure out the parsing.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by SavinayDharmappa** [comment]
Is this PR is merged ? I don't see change after pull.

File: scripts/extract_dts_includes.py (hunk 1 above)
Comment (galak) [inline scripts/extract_dts_includes.py]: What if we merge this as is, and figure out how to refactor extract_property later?

Comment (agross-linaro) [inline scripts/extract_dts_includes.py]: i have no problem with that.

**Comment by galak** [comment]
@SavinayDharmappa merged now

**Pull request merged**